R/seroprevalence_data.R

```diff
@@ -120,4 +120,203 @@ prepare_bin_data <- function(serodata) {
+#' This corresponds to the number of trials \code{size} in \link[stats]{rbinom}.
+#' @return Dataframe object containing the simulated data generated from \code{foi}
+#' @examples
+#'\dontrun{
+#' size_age_class = 5
+#' foi <- rep(0.02, 50)
+#' sim_data <- generate_sim_data(foi = foi,
+#'                               size_age_class = size_age_class,
+#'                               tsur = 2050,
+#'                               birth_year_min = 2000,
+#'                               survey_label = 'sim_constant_foi')
+#' }
+#' @export
+generate_sim_data <- function(foi,
+                              size_age_class,
+                              tsur,
+                              birth_year_min,
+                              survey_label,
+                              test = "fake",
+                              antibody = "IgG",
+                              seed = 1234
+                              ){
+    sim_data <- data.frame(birth_year = c(birth_year_min:(tsur - 1))) %>%
+        mutate(tsur = tsur,
+            country = 'None',
+            test = test,
+            antibody = antibody,
+            survey = survey_label,
+            age_mean_f = tsur - birth_year)
+    sim_data <- sim_data %>%
+        mutate(counts = get_sim_counts(sim_data, foi, size_age_class, seed = seed),
+            total = size_age_class) %>%
+        prepare_serodata(add_age_mean_f = FALSE)
+
+    return(sim_data)
+}
+
+#' Method for constructing age-group variable from age column
+#'
+#' This function was taken from \link[vaccineff]{get_age_group}.
+#' This method splits an age interval from min_val to max_val into
+#' (max_val-min_val)/step intervals.
+#' By default min_val is set 0, however it can be assigned by
+#' convenience.
+#' If the method finds ages greater or equal than max_val
+#' it assigns the string ">{max_val}".
+#' To avoid errors it is necessary to set step < max_val.
+#' It is also suggested to choose the step such
+#' that max_val%%(step+1) = 0.
+#' @param data dataset with at least a column containing the age
+#' information
+#' @param col_age name of the column containing the age
+#' information
+#' @param  max_val maximum value of age interval to split
+#' @param  min_val minimum value of age interval to split
+#' @param  step step used to split the age interval
+#' @return age_group
+#' @examples
+#' \dontrun{
+#' sim_data <- generate_sim_data(foi = rep(0.02, 50),
+#'                               size_age_class = 5,
+#'                               tsur = 2050,
+#'                               birth_year_min = 2000,
+#'                               survey_label = 'sim_constant_foi')
+#' sim_data$age_group <- get_age_group(data = sim_data,
+#'                                     col_age = "age_mean_f",
+#'                                     max_val = max(sim_data$age_mean_f),
+#'                                     min_val = min(sim_data$age_mean_f),
+#'                                     step = 5)
+#' }
+#' @export
```

Comment:
I think this is an internal function right? I can't imagine a user wanting this. So, I'd remove the `export`.

Reply:
Addressed in commit [31e147d](https://github.com/epiverse-trace/serofoi/pull/83/commits/31e147dfdd2eacfc94cc49d99318f9c4d6645709)